### DISCUSSION TITLE
How can I extract (existing) highlights in PDF?

### DISCUSSION THREAD

**Discussion by Kyeonghyeon-Park** [RyotaUshio/obsidian-pdf-plus#172] [Q&A] [answered]
I have several PDF files, which some sentences are already highlighted through Abobe PDF. 
I want to extract these highlights but I couldn't easily find the option among enormous features in PDF++. 
My ultimate goal is to use Obsidian for studying articles, so I want to extract existing highlights that I made and add notes on that. 
Thank you, in advance!

**Comment by RyotaUshio** [comment, marked as answer]
PDF++ has a command called "Extract & copy annotations in this PDF".

It will copy the annotations in the current PDF file according to the copy format (ex. "Quote in callout") & display text format (ex. "Title & page") specified in the toolbar.
To use a custom format, go to the PDF++ settings and create a new one as you wish.

**Reply by Kyeonghyeon-Park** [reply]
That's what I exactly want. Thank you. It seems that all features are useful and I love those, but I couldn't easily find a way to utilize them. I believe README and DOCS can be improved. I hope to fully utilize all features that you built!

**Reply by RyotaUshio** [reply]
Yeah, I know I need to update the docs but just can't find time for that... Sorry for the inconvenience!

**Reply by Kyeonghyeon-Park** [reply]
One quick question. Is there a way to extract highlights in the same order (reading perspective) in a double-column PDF? It seems that it orders the highlights with the location. For example, if highlights are located as below, it shows the highlights with the following order: C -> A -> D -> B. But I want to extract highlights with the order: A -> B -> C -> D. 
..  C
A D
B

**Reply by RyotaUshio** [reply]
It's currently not supported, but it's a good suggestion, thanks! It's now tracked at #176.

**Reply by lawriewalters63** [reply]
Can this be used for PDF++ highlight links? If not, how is this possible?
